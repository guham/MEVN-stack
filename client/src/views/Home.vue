<template>
  <div class="text-center">
    <img
      src="../assets/logo.png"
      alt=""
    >
    <HelloWorld msg="Welcome to Your Vue.js App" />
  </div>
</template>

<script>
import HelloWorld from '@/components/HelloWorld.vue';

export default {
  name: 'Home',
  metaInfo: {
    title: 'Home',
  },
  components: {
    HelloWorld,
  },
};
</script>
