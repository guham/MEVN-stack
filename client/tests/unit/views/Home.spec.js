import Home from '@/views/Home';
import factory from '../factory';

describe('Home.vue', () => {
  test('use HelloWorld component', () => {
    const wrapper = factory(Home);
    const { HelloWorld } = wrapper.vm.$options.components;
    expect(HelloWorld).toBeDefined();
  });

  test('has the expected html structure', () => {
    const wrapper = factory(Home);
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });
});
