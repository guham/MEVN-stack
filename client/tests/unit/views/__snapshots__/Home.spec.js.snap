// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue has the expected html structure 1`] = `
<div class="home"><img src="../assets/logo.png">
  <helloworld-stub msg="Welcome to Your Vue.js App"></helloworld-stub>
</div>
`;
